dashboard/test/integration/scripts_test.rb

```diff
@@ -68,4 +68,99 @@ class ScriptsTest < ActionDispatch::IntegrationTest
     follow_redirect!
     assert_response :success
   end
+
+  test 'levelbuilder instrictions for csp2-2020' do
+    Rails.application.config.stubs(:levelbuilder_mode).returns true
+    sign_in(create(:levelbuilder))
+
+    @unit = create :script, name: 'csp2-2020'
+    @lesson_group = create :lesson_group, script: @unit
+    @lockable_lesson = create(:lesson, script: @unit, name: 'Assessment Day', lockable: true, lesson_group: @lesson_group, has_lesson_plan: true, absolute_position: 9, relative_position: 9)
+    @level_group = create(:level_group, :with_sublevels, name: 'assessment 1')
+    @lockable_level_group_sl = create(:script_level, script: @unit, lesson: @lockable_lesson, levels: [@level_group], assessment: true)
+
+    get "/s/#{@unit.name}/instructions"
+    assert_response :success
+    assert_select '.instructions_summary', 1
+    assert_select '.instructions_summary h1', "CSP Unit 2 - The Internet ('20-'21)"
+    assert_select ".script_level", 1
+    assert_select '.script_level a[href="/s/csp2-2020/lessons/9/levels/1"]', 1
+    assert_select ".script_level a", "Level 1: assessment 1"
+  end
+
+  test 'levelbuilder instructions for multiple lesson types' do
+    Rails.application.config.stubs(:levelbuilder_mode).returns true
+    sign_in(create(:levelbuilder))
+
+    @unit = create :script, name: 'csp2-2020'
+    lesson_group = create :lesson_group, script: @unit
+    @lesson = create(
+      :lesson,
+      script_id: @unit.id,
+      lesson_group: lesson_group,
+      name: 'first lesson',
+      relative_position: 1,
+      absolute_position: 1,
+      has_lesson_plan: true,
+      properties: {
+        overview: 'lesson overview',
+        student_overview: 'student overview'
+      }
+    )
+    @lesson2 = create(
+      :lesson,
+      script_id: @unit.id,
+      lesson_group: lesson_group,
+      name: 'second lesson',
+      relative_position: 2,
+      absolute_position: 2,
+      has_lesson_plan: true
+    )
+    @level = create :maze
+    @level2 = create(:level, long_instructions: 'foo', short_instructions: 'bar')
+    @level3 = create :multi
+    @script_level1 = create(
+      :script_level,
+      activity_section: @activity_section,
+      activity_section_position: 1,
+      chapter: 1,
+      position: 1,
+      lesson: @lesson,
+      script: @lesson.script,
+      levels: [@level],
+      challenge: true
+    )
+    @script_level2 = create(
+      :script_level,
+      activity_section: @activity_section,
+      activity_section_position: 1,
+      chapter: 1,
+      position: 2,
+      lesson: @lesson,
+      script: @lesson.script,
+      levels: [@level2],
+      challenge: true
+    )
+    @script_level3 = create(
+      :script_level,
+      activity_section: @activity_section,
+      activity_section_position: 1,
+      chapter: 1,
+      position: 1,
+      lesson: @lesson2,
+      script: @lesson2.script,
+      levels: [@level3],
+      challenge: true
+    )
```

Comment:
you may be able to replace all of this with something like:
```suggestion
@unit = create(:script, name: ..., :with_levels, lessons_count: 2, levels_count: 2)
@unit.lessons.first.script_levels.first.update!(challenge: true)
...
assert_select '.instructions_summary h1', "Lesson 1: #{@script.lessons.first.name}"
```
this invokes: https://github.com/code-dot-org/code-dot-org/blob/9a47018ebbef4976863527c443839b8b9f9b6263/dashboard/test/factories/factories.rb#L940-L952

fine to keep what you have this time, but worth looking in the factories for shortcuts next time.